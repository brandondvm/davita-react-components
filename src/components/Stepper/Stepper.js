import React from 'react';
import PropTypes from 'prop-types';

const Stepper = ({ steps = 0, activeStep = 0 }) => {
  const stepsArray = [ ...Array(steps).keys() ];
  return (
    <div className="stepper-component">
      <div className="stepper-component_line"></div>
      {
        stepsArray.map((step, index) =>
          <div key={step} className={`stepper-component_step ${(index + 1) <= activeStep ? 'stepper-component_step_active' : ''}`} />
        )
      }      
    </div>
  );
};

Stepper.propTypes = {
  steps: PropTypes.number,
  activeStep: PropTypes.number
};

export default Stepper;
