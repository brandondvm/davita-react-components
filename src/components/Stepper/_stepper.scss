.stepper-component {
  align-items: center;
  display: flex;
  justify-content: space-between;
  position: relative;

  &_step {
    align-items: center;
    background: $munsell;
    border-radius: 30px;
    display: flex;
    height: 24px;
    justify-content: center;
    position: relative;
    width: 24px;

    &:last-child {
      margin-right: 0;
      &::after {
        content: none;
      }
    }

    &_active::before {
      background: $lightblue;
      border-radius: 30px;
      content: '';
      height: 16px;
      width: 16px;
    }
  }

  &_line {
    background: $munsell;
    height: 2px;
    position: absolute;
    width: 99%;
  }
}
