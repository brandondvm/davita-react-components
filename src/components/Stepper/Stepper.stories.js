import React from 'react';

import Stepper from './Stepper';
import { number } from 'prop-types';

export default {
  title: 'Stepper',
  component: Stepper,
  argTypes: {
    steps: number,
    activeStep: number
  },
};

const Template = (args) => <div className="storybook-wrapper"><Stepper {...args} /></div>;

export const Default = Template.bind({});
Default.args = {
  steps: 6,
  activeStep: 0,
};

export const Active = Template.bind({});
Active.args = {
  steps: 6,
  activeStep: 3,
};
