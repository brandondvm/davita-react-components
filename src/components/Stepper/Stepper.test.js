import enzyme from 'enzyme';
import React from 'react';

import Stepper from './Stepper';

const mockProps = {
  steps: 6,
  activeStep: 3,
};

function setup(passedProps = {}) {
  const updatedProps = {
    ...mockProps,
    ...passedProps,
  };

  return enzyme.shallow(<Stepper { ...updatedProps } />);
}

describe('Stepper Component', () => {
  test('Component renders correctly', () => {
    const wrapper = setup();
    expect(wrapper.get(0)).toMatchSnapshot();
  });

  test('Component should render 6 steps when steps prop is 6', () => {
    const wrapper = setup();
    const activeSteps = wrapper.find('.stepper-component_step');
    expect(activeSteps.length).toEqual(6);
  });

  test('Component should render 3 steps when steps prop is 3', () => {
    const wrapper = setup({ steps: 3 });
    const activeSteps = wrapper.find('.stepper-component_step');
    expect(activeSteps.length).toEqual(3);
  });

  test('Component should has 3 active steps when active step is 3', () => {
    const wrapper = setup();
    const activeSteps = wrapper.find('.stepper-component_step_active');
    expect(activeSteps.length).toEqual(3);
  });

  test('Component should has 0 active steps when is not active step', () => {
    const wrapper = setup({ activeStep: 0 });
    const activeSteps = wrapper.find('.stepper-component_step_active');
    expect(activeSteps.length).toEqual(0);
  });
});
