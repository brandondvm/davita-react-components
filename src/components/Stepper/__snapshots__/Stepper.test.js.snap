// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper Component Component renders correctly 1`] = `
<div
  className="stepper-component"
>
  <div
    className="stepper-component_line"
  />
  <div
    className="stepper-component_step stepper-component_step_active"
  />
  <div
    className="stepper-component_step stepper-component_step_active"
  />
  <div
    className="stepper-component_step stepper-component_step_active"
  />
  <div
    className="stepper-component_step "
  />
  <div
    className="stepper-component_step "
  />
  <div
    className="stepper-component_step "
  />
</div>
`;
